coap/adapter.go

```diff
@@ -70,17 +77,7 @@ func (svc *adapterService) Publish(ctx context.Context, key string, msg messagin
 	}
 	msg.Publisher = thid.GetValue()
 
-	data, err := proto.Marshal(&msg)
-	if err != nil {
-		return err
-	}
-
-	subject := fmt.Sprintf("%s.%s", chansPrefix, msg.Channel)
-	if msg.Subtopic != "" {
-		subject = fmt.Sprintf("%s.%s", subject, msg.Subtopic)
-	}
```

Comment:
Are you sure about removing subtopic here?

Reply:
This is already implemented in the nats wrapper [here](https://github.com/mainflux/mainflux/blob/5ca8495f35691b7c97736cedc873168fb1f7688f/pkg/messaging/nats/pubsub.go#L73)

When you pass the topic and the message it will extract the subtopic from the message as implemented [here](https://github.com/mainflux/mainflux/blob/5ca8495f35691b7c97736cedc873168fb1f7688f/pkg/messaging/nats/pubsub.go#L74)

Reply:
Ok, thank you for exaplaining!